contrib/platform/test/com/sun/jna/platform/mac/DiskArbitrationTest.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.mac;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.mac.CoreFoundation.CFBooleanRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFDictionaryRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFMutableDictionaryRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFNumberRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFStringRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFTypeRef;
+import com.sun.jna.platform.mac.DiskArbitration.DADiskRef;
+import com.sun.jna.platform.mac.DiskArbitration.DASessionRef;
+import com.sun.jna.platform.mac.IOKit.IOIterator;
+import com.sun.jna.platform.mac.IOKit.IORegistryEntry;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.PointerByReference;
+
+public class DiskArbitrationTest {
+
+    private static final DiskArbitration DA = DiskArbitration.INSTANCE;
+    private static final CoreFoundation CF = CoreFoundation.INSTANCE;
+    private static final IOKit IO = IOKit.INSTANCE;
+    private static final SystemB SYS = SystemB.INSTANCE;
+
+    @Test
+    public void testDiskCreate() {
+        IntByReference masterPortPtr = new IntByReference();
+        assertEquals(0, IO.IOMasterPort(0, masterPortPtr));
+        int masterPort = masterPortPtr.getValue();
+
+        // Create some keys we'll need
+        CFStringRef daMediaBSDName = CFStringRef.createCFString("DAMediaBSDName");
+        CFStringRef daMediaWhole = CFStringRef.createCFString("DAMediaWhole");
+        CFStringRef daMediaLeaf = CFStringRef.createCFString("DAMediaLeaf");
+        CFStringRef daMediaSize = CFStringRef.createCFString("DAMediaSize");
+        CFStringRef daMediaBlockSize = CFStringRef.createCFString("DAMediaBlockSize");
+
+        // Open a DiskArbitration session
+        DASessionRef session = DA.DASessionCreate(CF.CFAllocatorGetDefault());
+        assertNotNull(session);
+
+        // Get IOMedia objects representing whole drives and save the BSD Name
+        List<String> bsdNames = new ArrayList<>();
+        PointerByReference iterPtr = new PointerByReference();
+
+        CFMutableDictionaryRef dict = IOKit.INSTANCE.IOServiceMatching("IOMedia");
+        // Consumes a reference to dict
+        assertEquals(0, IO.IOServiceGetMatchingServices(masterPort, dict, iterPtr));
+        IOIterator iter = new IOIterator(iterPtr.getValue());
+        IORegistryEntry media = iter.next();
+        while (media != null) {
+            CFStringRef wholeKey = CFStringRef.createCFString("Whole");
```

Comment:
You could pull this out of the loop and add it to the constant strings lines 66-70.